controllers/prep_handlers.go

```diff
@@ -97,28 +124,40 @@ func (r *ImageBasedUpgradeReconciler) checkSeedImageCompatibility(_ context.Cont
 
 	// TODO: use the context when execute supports it
 	if inspectRaw, err := r.Executor.Execute("podman", inspectArgs...); err != nil || inspectRaw == "" {
-		return fmt.Errorf("failed to inspect image: %w", err)
+		return nil, fmt.Errorf("failed to inspect image: %w", err)
 	} else {
 		if err := json.Unmarshal([]byte(inspectRaw), &inspect); err != nil {
-			return fmt.Errorf("failed to unmarshal image inspect output: %w", err)
+			return nil, fmt.Errorf("failed to unmarshal image inspect output: %w", err)
 		}
 	}
 
 	if len(inspect) != 1 {
-		return fmt.Errorf("expected 1 image inspect result, got %d", len(inspect))
+		return nil, fmt.Errorf("expected 1 image inspect result, got %d", len(inspect))
 	}
 
-	seedFormatLabelValue, ok := inspect[0].Labels[common.SeedFormatOCILabel]
+	return inspect[0].Labels, nil
+}
+
+// checkSeedImageVersionCompatibility checks if the seed image is compatible with the
+// current version of the lifecycle-agent by inspecting the OCI image's labels
+// and checking if the specified format version equals the hard-coded one that
+// this version of the lifecycle agent expects. That format version is set by
+// the LCA during the image build process to the value of the code constant,
+// and the code constant is only manually bumped by developers when the image
+// format changes in a way that is incompatible with previous versions of the
+// lifecycle-agent.
+func checkSeedImageVersionCompatibility(labels map[string]string) error {
+	seedFormatLabelValue, ok := labels[common.SeedFormatOCILabel]
 	if !ok {
 		return fmt.Errorf(
-			"seed image %s is missing the %s label, please build a new image using the latest version of the lca-cli",
-			seedImageRef, common.SeedFormatOCILabel)
+			"seed image is missing the %s label, please build a new image using the latest version of the lca-cli",
```

Comment:
Not necessarily "latest version", but seed was generated from "incompatible version of LCA" maybe?

Reply:
I agree, but unrelated to PR. Created #423